### ISSUE TITLE
[sbpp_comms.smx] Query_ProcessQueue failed: malformed database schema (?)

### ISSUE THREAD

**Issue by unusual917** [sbpp/sourcebans-pp#478]
What are the steps to reproduce this issue?
-------------------------------------------
1. Install sourceban and check sourcemod logs.

What happens?
-------------
Error shows [sbpp_comms.smx] Query_ProcessQueue failed: malformed database schema (?)
The plugin works fine however. I do notice map change is slow due to database hanging on map change.

What were you expecting to happen?
----------------------------------
Run without this error.

Any logs, error output, etc.?
----------------------------
[sbpp_comms.smx] Query_ProcessQueue failed: malformed database schema (?)

Any other comments?
-------------------
How can I troubleshoot this and find the cause?

What versions of software are you using?
----------------------------------------
**Operating System:**
Debian 7
**SourceBans++ Version:**
Latest version.
**SourceMod Version:**
Latest version.
**MetaMod Version:**
Latest version.

**Comment by Groruk** [comment]
> What are the steps to reproduce this issue?
>--------------------------------------------
>1. Install sourceban and check sourcemod logs.

Got me a working installation of SourceBans++ and no errors in the Sourcemod log.

>SourceBans++ Version:
Latest version.

Which 'latest' version of SourceBans++? 1.6.3 stable or 1.6.4 dev?

>SourceMod Version:
Latest version.

Again latest stable (1.8) or latest dev (1.9)? Just write your version number...

>MetaMod Version:
Latest version.

1.10 or 1.11?

>How can I troubleshoot this and find the cause?

1. `[sbpp_comms.smx]` you know the plugin so you can lookup the [source](https://github.com/sbpp/sourcebans-pp/blob/v1.x/game/addons/sourcemod/scripting/sbpp_comms.sp).
2. `Query_ProcessQueue` you also know the function which causes the issue [#L1549](https://github.com/sbpp/sourcebans-pp/blob/v1.x/game/addons/sourcemod/scripting/sbpp_comms.sp#L1549)
3. [This code block](https://github.com/sbpp/sourcebans-pp/blob/v1.x/game/addons/sourcemod/scripting/sbpp_comms.sp#L1551-L1555) shows you that either there is a `INVALID_HANDLE` or an `error`
4. Find out where `Query_ProcessQueue` is [called](https://github.com/sbpp/sourcebans-pp/blob/v1.x/game/addons/sourcemod/scripting/sbpp_comms.sp#L1230-L1232). This shows that `Query_processQueue` is a callback for a `SQL Query`. In this case a `SQLite Query`. We can also assume that this is a SQLite issue.
5. [Googling](https://media.giphy.com/media/3oKIPmM4hkQIf4p5T2/giphy.gif) the actual error `malformed database schema` confirms that this is a SQLite issue. Also first link is a blog post about how to [potentially fix the problem](https://www.arysontechnologies.com/blog/fix-sqlite-error-malformed-database-schema/).
6. Now we need to find out where the `SQLite Handle` is created [#L2047](https://github.com/sbpp/sourcebans-pp/blob/v1.x/game/addons/sourcemod/scripting/sbpp_comms.sp#L2047)
7. We now know that the `SQLlite database` is located in `addons/sourcemod/data/sqlite/sourcecomms-queue.sq3`.
8. [This Query](https://github.com/sbpp/sourcebans-pp/blob/v1.x/game/addons/sourcemod/scripting/sbpp_comms.sp#L2053-L2063) seems to not cause any issues so maybe the SQLite database got corrupted during creation or while trying to create a new table. Since no other SQLite database is causing problems this can be ruled an anomaly.
9. Best bet would be to shutdown the gameserver, delete the existing `sourcecomms-queue.sq3` and start the server to see if the database was created successfully this time.

**Comment by CrazyHackGUT** [comment]
> latest dev (1.9)

Latest dev - 1.10

**Comment by unusual917** [comment]
Hi Groruk, I'm using the latest stable, thought the dev versions would cause issues.
Thank you for the detailed look at the issue btw! I googled it as well and thought it was odd sqlite was being used when I'm strictly using MySQL. The functions work and data pushes to sourcebans++ webpanel, but I get this error still and was worried.

Any idea why it's using both sqlite and MySQL?
I'll try deleting sourcecomms-queue.sq3. (Just tried and tried restarting as well. Interesting how the file isn't recreated now. I tried checking the file permissions also).

Thanks again.
